rclcpp_lifecycle/test/test_lifecycle_node.cpp

```diff
@@ -142,6 +150,22 @@ TEST_F(TestDefaultStateMachine, empty_initializer) {
   EXPECT_EQ(State::PRIMARY_STATE_UNCONFIGURED, test_node->get_current_state().id());
 }
 
+TEST_F(TestDefaultStateMachine, empty_initializer_rcl_errors) {
+  {
+    auto patch = mocking_utils::patch_and_return(
+      "lib:rclcpp_lifecycle", rcl_lifecycle_state_machine_init, RCL_RET_ERROR);
+    EXPECT_THROW(
+      std::make_shared<EmptyLifecycleNode>("testnode"),
+      std::runtime_error);
+  }
+  {
+    auto test_node = std::make_shared<EmptyLifecycleNode>("testnode");
+    auto patch = mocking_utils::inject_on_return(
+      "lib:rclcpp_lifecycle", rcl_lifecycle_state_machine_fini, RCL_RET_ERROR);
+    EXPECT_NO_THROW(test_node.reset());
```

Comment:
Is it there anyway to check that this caused an error? looks like expected behavior to me

Reply:
In #1297 we decided to make `reset()` noexcept, which is called during destruction (not to be confused with the one called here in line 165 since test_node is a `shared_ptr`). Therefore if `rcl_lifecycle_state_machine_fini` fails, destruction will just continue silently and an error will be logged.